### ISSUE TITLE
Ids are not added to the base resource endpoint

### ISSUE THREAD

**Issue by leandroz** [js-data/js-data-localforage#3]
Hey!
I just started using this adapter for offline support for a particular resource, not the entire store. 
I am:
- Registering the adapter (the default is a custom adapter)
````
var localForageAdapter = new DSLocalForageAdapter();
DS.adapters.localforage = localForageAdapter;
````
- Then in afterInject hook:
````
			afterInject(resource, data) {
				if (data.length) {
					angular.forEach(data, (post) => {
						if (post.post_id) {
							DS.adapters.localforage.create(resource, post);
						}
					});
				} else if (data.post_id) {
					DS.adapters.localforage.create(resource, data);
				}
				return;
			}
````
- And then getting all the posts with:
````
				DS.adapters.localforage.findAll(Post)
					.then((posts) => {
						console.log(posts);
					});
````
The problem is that i am not getting all the posts that are stored because in the "/post/" key not all of them are added. I fist get the posts with Post.findAll({params}), make some changes and then Post.inject for updating the in memory store.
![problemwithlocalforage](https://cloud.githubusercontent.com/assets/4624850/7111526/150911ea-e204-11e4-8f00-8321f021cbd1.png)

**Comment by jmdobry** [comment]
Hmm, I think there might be some race conditions that happen when you loop over the items and save them individually (but in parallel). What happens when you do this:

```js
afterInject(resource, data) {
  if (data.length) {
    let promise;
    angular.forEach(data, (post) => {
        if (post.post_id) {
          if (promise) {
            promise = promise.then(() => {
              return DS.adapters.localforage.create(resource, post);
            });
          } else {
            promise = DS.adapters.localforage.create(resource, post);
          }
        }
    });
  } else if (data.post_id) {
    DS.adapters.localforage.create(resource, data);
  }
  return;
}
```

This will help me confirm my suspicions of a race condition.

**Comment by leandroz** [comment]
Hey @jmdobry, thanks for the quick response. Is working now, I think is because I was trying to create a new one while it was still an operation in progress so your suspicions were correct.

I am looking another issue.
I have posts, dogs and users, a post has one user and one dog by the user_id and dog_id relations. I am creating the posts, users and dogs in the afterInject hook of each one into local forage using the same as above. Sometimes I make a manual User.inject or Dog.inject that doesn't came from a find or findAll in the resource. The user and dog are created into local forage but the /dog and /user are not updated with that id. The same as this issue but for a different cause.
The ones that are because of the post relations and findAll or find methods in the stores are created ok, the ones injected manually don't.

Let me know if you need more information.

**Comment by jmdobry** [comment]
A plunker that demonstrates the issue would be absolutely the most helpful thing. Here's one you can start from: http://plnkr.co/edit/yT5DZO

**Comment by leandroz** [comment]
Is actually the same problem. 
The afterInject on the User resource is called because of the relationship with the Post resource (the post object has post.user with the user object), the same for Dog.
I think that is working as a loop where the first promise is not finished so for some reason is not added to the /user or /dog key but the /user/:user_id and /dog/:dog_id is actually created.
Let me know when you fix the bug or if i can help you with that.

**Comment by jmdobry** [comment]
@leandroz The best thing you can do is create a working example (using [this plunker](http://plnkr.co/edit/yT5DZO) as base) that I can use to swiftly debug the issue. I haven't even seen your resource definitions, so at this point fixing it would take 10x longer than if I had an example that demonstrates the issue.

**Comment by leandroz** [comment]
Here is the plunker, http://plnkr.co/edit/CZri1MrEv4jGVdIogJ2t?p=info, it has the relations and everything, see that the /dog and /user endpoints only has the last item in IndexedDB.

**Comment by jmdobry** [comment]
Thanks

**Comment by leandroz** [comment]
@jmdobry i am facing problems with the relations too.
When i define the User and Dog resources, I am doing:
````
		dogResource.findAll({}, {
			adapter: 'localforage'
		})
			.then(dogs => {
				console.log(dogs);
				console.log(dogResource.filter({}));
			});
````
The same with the User to load the data from local forage  into the store.
After that:
```
					Post.findAll({}, {
						adapter: 'localforage'
					})
						.then(posts => {
							console.log('POSTS FROM localforage');
							console.log(posts);
                            Post.linkAll({}, ['user', 'dog']);
							console.log('POST WITH LINKED RELATIONS');
							console.log(Post.filter({}));
							deferred.resolve();
						});
````
But the relations are never loaded. The definitions are the same ones in the plunker.

**Comment by jmdobry** [comment]
Regarding the relations not getting loaded, I believe you found a bug in js-data. If you change the Post `hasOne` relation to a `belongsTo` relation then it will work. I'll have a look at js-data to see why it doesn't work for a `hasOne` relation.

**Comment by jmdobry** [comment]
The relation thing will be fixed in js-data 2.0. The race condition will be fixed in js-data-localforage 2.0. See js-data/js-data#107 #4

**Comment by leandroz** [comment]
Great news @jmdobry !, when do you think that version 2.0 will be ready?, so I know if a need to patch the errors or I can wait, let me know if you need any help!

**Comment by jmdobry** [comment]
js-data 2.0.0-beta.3 and js-data-localforage 2.0.0-beta.1 are available in NPM, so you can start testing now (which would be appreciated). For Js-data 2.0, breaking changes should be documented in its changelog.

I don't have plans for any other changes to js-data-localforage before I release 2.0, but I still have some features I need to figure out for js-data 2.0 before I can release it. How long will that take? Couple of weeks maybe.

**Comment by jmdobry** [comment]
Also, I tested your plunker with js-data-localforage 2.0.0-beta.1 and everything seemed to work (including changing the `hasOne` to `belongsTo`)

**Comment by leandroz** [comment]
I updated both and everything is working perfect!, thanks @jmdobry great work!
